consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -718,3 +744,97 @@ impl<
         counters::CUR_SLOT_BALLOT_COUNTER.set(slot_metrics.bN as i64);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        byzantine_ledger::{
+            ledger_sync_state::LedgerSyncState,
+            tests::{get_local_node_config, get_peers},
+            worker::ByzantineLedgerWorker,
+        },
+        tx_manager::MockTxManager,
+    };
+    use mc_common::logger::{test_with_logger, Logger};
+    use mc_connection::ConnectionManager;
+    use mc_consensus_scp::{MockScpNode, QuorumSet};
+    use mc_ledger_db::MockLedger;
+    use mc_ledger_sync::MockLedgerSync;
+    use mc_peers::{ConsensusMsg, MockBroadcast};
+    use mc_peers_test_utils::MockPeerConnection;
+    use mc_util_metrics::OpMetrics;
+    use rand::rngs::StdRng;
+    use rand_core::SeedableRng;
+    use std::sync::{
+        atomic::{AtomicBool, AtomicU64},
+        Arc, Mutex,
+    };
+
+    #[test_with_logger]
+    /// Test that `new` correctly initializes the instance.
+    fn test_new(logger: Logger) {
```

Comment:
This test is mostly a proof-of-concept showing how to unit test the worker using mocks.